src/utils.js

```diff
@@ -753,6 +753,49 @@ export function toIdSchema(
   return idSchema;
 }
 
+export function toNameSchema(schema, name = "", definitions, formData = {}) {
```

Comment:
This is great -- and we can later pass on the `nameSchema` to all widgets if we need them to have a name. Would it be more appropriate to call it `pathSchema` instead though?

Additionally, given https://github.com/mozilla-services/react-jsonschema-form/pull/1248, could we potentially namespace `$name`?

Also, how would we distinguish an array element from an actual object property called "0" or "1" or "2"?

Reply:
Ya, I think that's a good idea. 

Also, you are right about those array indexes. I can rename them to be like "$0", "$1", "$2" ...

Reply:
@epicfaace got some time to work on this now, the namespace thing you guys were talking about -- that is something that doesn't exist currently, right? We could potentially namespace  `$name` or something similar in the future.

As for the array indexes, I think even having "$0", "$1", etc. would be too common. What about something crazy for now like having escaped quotes? 

For example:

```
{
  list: {
    '0': { ... },
    '1': { ... },
    '2': { ... }
}
```

Would it be possible to namespace something like the array indexes? `somePrefix+number`?

Reply:
Just updated this comment because I was thinking about it I changed my mind and don't think it would be a good idea to differentiate array indexes from object properties named like "0", "1", "2", etc. because they are handled the same way when using lodash `pick`. In the future, if someone needed to use the name/path schema for something else and wanted to know if a number path was an array index or property, then they could check it against the schema or form data.